lib/build.gradle.kts

```diff
@@ -0,0 +1,100 @@
+plugins {
+    id("com.android.library")
+    id("org.jetbrains.kotlin.android")
+    id("org.jetbrains.dokka")
+    id("maven-publish")
+}
+
+android {
+    namespace = "at.bitfire.vcard4android"
+
+    compileSdk = 33
+
+    defaultConfig {
+        minSdk = 21        // Android 5
+        targetSdk = 33     // Android 13
+
+        aarMetadata {
+            minCompileSdk = 29
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+
+        isCoreLibraryDesugaringEnabled = true
+    }
+    kotlin {
+        jvmToolchain(17)
+    }
+
+    buildFeatures {
+        buildConfig = true
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+        }
+    }
+
+    packagingOptions {
+        resources {
+            excludes += listOf("LICENSE", "META-INF/LICENSE.txt", "META-INF/NOTICE.txt")
+        }
+    }
+
+    lint {
+        disable += listOf("AllowBackup", "InvalidPackage")
+    }
+
+    defaultConfig {
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    publishing {
+        // Configure publish variant
+        singleVariant("release") {
+            withSourcesJar()
+        }
+    }
+}
+
+publishing {
+    // Configure publishing data
+    publications {
+        register("release", MavenPublication::class.java) {
+            groupId = "com.github.bitfireAT"
+            artifactId = "vcard4android"
+            version = project.properties["vcard4android.version"] as String
```

Comment:
Is it possible to omit this version number or use the commit ID instead?

Because I suspect that we won't introduce proper vcard4android versioning in the next future and when all these versions are always shown as "1.0.0" it might be confusing …

Reply:
We can add a compiler argument for this, I suppose. We just need to update the workflow

Reply:
Hm… sounds cumbersome, too… or can we use something like "snapshot" (or "0.0" forever) as gradle version? We decide the real version over the dependency anyway

Reply:
Okay, so I have found out that Jitpack has an environment variable for the compiling commit ID. Since you can pick which commit you want to get (just like selecting a version), maybe we can use this approach.

Reply:
I think that would be a good solution :)